app/src/main/java/com/example/tasky/feature_agenda/data/repository/EventRepositoryImpl.kt

```diff
@@ -2,217 +2,144 @@ package com.example.tasky.feature_agenda.data.repository
             }
+        } catch (e: HttpException) {
+            return Resource.Error(e.message ?: "Invalid Response")
+        } catch (e: IOException) {
+            return Resource.Error(e.message ?: "Couldn't reach server")
         }
-        return Result.Success(validatedAttendees)
     }
 
+
     @RequiresApi(Build.VERSION_CODES.O)
     override suspend fun createEvent(
         event: AgendaItem.Event
-    ): Result<Unit> {
-
-       return try {
-
-           val photoList = event.photos.map { photo ->
-               val file = File(photo.url)
-               val requestFile: RequestBody = file.asRequestBody("image/*".toMediaTypeOrNull())
-               MultipartBody.Part.createFormData("photos", file.name, requestFile)
-           }
-
-           var validatedAttendees = listOf<Attendee>()
-           val doAttendeesExist = validateAttendees(event.attendees)
-
-           when(doAttendeesExist) {
-               is Result.Error -> {
-                   return Result.Error(doAttendeesExist.message ?: "Attendees not found!")
-               }
-               is Result.Success -> {
-                   validatedAttendees = doAttendeesExist.data ?: return Result.Error("Something went wrong!")
-               }
-           }
-
-            api.createEvent(
-                EventRequest(
-                    id = event.eventId,
-                    title = event.eventTitle,
-                    description = event.eventDescription ?: "",
-                    from = event.from.toUtcTimestamp(),
-                    to = event.to.toUtcTimestamp(),
-                    remindAt = event.remindAt.toUtcTimestamp(),
-                    attendeeIds = validatedAttendees.map { it.userId }
-                ),
-                photoList
-            )
-
-           val eventEntity = event.toEventEntity()
-           val attendeeEntities = validatedAttendees.map { it.toAttendeeEntity(event.eventId) }
-
-           db.agendaDao.upsertEventWithAttendees(db, eventEntity, attendeeEntities)
-           
-           Result.Success()
-
-        } catch(e: HttpException) {
-           Result.Error(e.message ?: "Invalid Response")
-        } catch(e: IOException) {
-           Result.Error(e.message ?: "Couldn't reach server")
-        }
-    }
+    ): Resource<Unit> {
 
-    @RequiresApi(Build.VERSION_CODES.O)
-    override suspend fun getEvent(eventId: String): Result<AgendaItem.Event> {
+        val eventEntity = event.toEventEntity()
+        val attendeeEntities = event.attendees.map { it.toAttendeeEntity(event.eventId) }
 
-        val localEvent = db.eventDao.getEventWithAttendees(eventId.toInt())
+        db.agendaDao.upsertEventWithAttendees(db, eventEntity, attendeeEntities)
 
-        localEvent?.let {
-            return Result.Success(localEvent.toEvent())
-        }
+        val jsonEvent = moshi.adapter(AgendaItem.Event::class.java).toJson(event)
+        val inputData = Data.Builder()
+            .putString("action", "create")
```

Comment:
For more consistency, I'd work with an enum here to avoid typos and needing to look up the allowed values